task/sast-coverity-check/0.1/sast-coverity-check.yaml

```diff
@@ -0,0 +1,263 @@
+        else
+            echo "Coverity capture command failed with exit code ${COV_CAPTURE_EXIT_CODE}. Exiting..."
+            exit 1
+        fi
+        
+        # Analysis phase
+        /opt/coverity/bin/cov-manage-emit --dir $COVERITY_DIR reset-host-name
+        /opt/coverity/bin/cov-analyze $COV_ANALYZE_ARGS --dir=$COVERITY_DIR
+        COV_ANALYZE_EXIT_CODE=$?
+        
+        if [[ "$COV_ANALYZE_EXIT_CODE" -eq 0 ]]; then
+          echo "Coverity analyze scan finished successfully"
+        else
+          echo "Coverity analyze scan failed with exit code ${COV_ANALYZE_EXIT_CODE}. Exiting..."
+          exit 1
+        fi
+        set -e
+        
+        /opt/coverity/bin/cov-format-errors --dir=$COVERITY_DIR --json-output-v10 $COVERITY_RESULTS_FILE
+        # We parse the results, embed context, remove duplicates and store them in SARIF format.
+        IMP_LEVEL=1
+        if [[ "${IMP_FINDINGS_ONLY}" == "false" ]]; then
+          IMP_LEVEL=0
+        fi
+        
+        (cd "$SOURCE_CODE_DIR" && csgrep --mode=json --imp-level="$IMP_LEVEL" --remove-duplicates --embed-context=3 $COVERITY_RESULTS_FILE) \
+          | csgrep --mode=json --strip-path-prefix="$SOURCE_CODE_DIR"/source/ \
+          | csgrep --mode=json --strip-path-prefix="/var/tmp/coverity/home" \
+          > sast_coverity_buildless_check_all_findings.json
+        
+        echo "Results:"
+        (set -x; csgrep --mode=evtstat sast_coverity_buildless_check_all_findings.json)
+
+        # We check if the KFP_GIT_URL variable is set to apply the filters or not
+        if [[ -z "${KFP_GIT_URL}" ]]; then
+          mv sast_coverity_buildless_check_all_findings.json filtered_sast_coverity_buildless_check_all_findings.json
+        else
+          echo "Filtering false positives in results files using csfilter-kfp..."
+          CMD=(
+              csfilter-kfp
+              --verbose 
+              --kfp-git-url="${KFP_GIT_URL}"
+          )
+          if [[ -n "${PROJECT_NVR}" ]]; then
+            CMD+=(--project-nvr="${PROJECT_NVR}")
+          fi
+          if [ "${RECORD_EXCLUDED}" == "true" ]; then
+            CMD+=(--record-excluded="excluded-findings.json")
+          fi
+          
+          "${CMD[@]}" sast_coverity_buildless_check_all_findings.json > filtered_sast_coverity_buildless_check_all_findings.json
+          status=$?
+          if [ "$status" -ne 0 ]; then
+            echo "Error: failed to filter known false positives" >&2
+            return 1
+          else
+            echo "Message: Succeed to filter known false positives" >&2
+          fi
+
+          echo "Results after filtering:"
+          (set -x; csgrep --mode=evtstat filtered_sast_coverity_buildless_check_all_findings.json)
+        fi
+
+        csgrep --mode=sarif filtered_sast_coverity_buildless_check_all_findings.json > "$(workspaces.workspace.path)"/hacbs/"$(context.task.name)"/coverity-buildless-results.sarif
```

Comment:
I would remove the `-buildless` substring from the uploaded results file so that consumers expecting a particular name do not need to update their scripts once we extend the task to do buildful scanning, too.

Reply:
Removed